compiler/rustc_hir/src/diagnostic_items.rs

```diff
@@ -0,0 +1,9 @@
+use crate::def_id::DefId;
+use rustc_data_structures::fx::FxHashMap;
+use rustc_span::Symbol;
+
+#[derive(Debug, Default, HashStable_Generic)]
```

Comment:
@oli-obk Without much understanding of queries, it seems like something is suboptimal if this struct is `HashStable_Generic`? Because only one of the fields should be hashed.

Reply:
hmm... good catch. it could be a perf issue, we can perf it, and if it is, implement HashStable manually?

Reply:
Another thing I considered is to break it into two queries. But I don't know if you can define a query that runs once and is only cached in memory?

Reply:
idk, my initial reaction is that another query is overkill for this. I think I would rather use a manual HashStable impl